src/App.tsx

```diff
@@ -1,183 +1,74 @@
-  return (
-    <>
-      <div className={styles.main}>
-        <ReferendaDeck
-          network={networkFor(api)}
-          referenda={referenda}
-          tracks={tracks}
-          voteHandler={voteHandler}
-        />
-      </div>
-      {referenda.length > 0 && (
-        <ActionBar
-          left={referenda.length}
-          onAccept={() =>
-            topReferenda &&
-            voteHandler(topReferenda, createStandardAccountVote(true))
-          }
-          onRefuse={() =>
-            topReferenda &&
-            voteHandler(topReferenda, createStandardAccountVote(false))
-          }
+export function App(): JSX.Element {
+  const [state, updater] = useLifeCycle();
+
+  switch (state.type) {
+    case 'InitialState':
+      return <div>INITIAL</div>;
+    case 'RestoredState':
+      return <div>RESTORED</div>;
+    case 'ConnectingState':
+      return <div>CONNECTING</div>;
+    case 'OfflineState':
+      return <div>OFFLINE</div>;
+    case 'SyncingState':
+      return (
+        <LoadingPanel
+          message={`Get ready to vote! #${JSON.stringify(state.since)}`}
         />
-      )}
-    </>
-  );
-}
-
-enum State {
-  LOADING,
-  STARTED,
-}
-
-type StartedContext = {
-  state: State.STARTED;
-  tracks: Map<number, Track>;
-  referenda: Map<number, ReferendumOngoing>;
-  accountVotes: Map<number, AccountVote>;
-};
-
-export type StateContext = { state: State.LOADING } | StartedContext;
-
-function voteOn(
-  index: number,
-  vote: AccountVote,
-  setStateContext: React.Dispatch<React.SetStateAction<StartedContext>>
-) {
-  setStateContext((stateContext) => {
-    stateContext.accountVotes.set(index, vote);
-    return { ...stateContext }; // Copy object to trigger re-draw
-  });
-}
-
-function AppPanel({
-  api,
-  connectedAccount,
-  context,
-  voteHandler,
-}: {
-  api: ApiPromise;
-  connectedAccount: SigningAccount | undefined;
-  context: StateContext;
-  voteHandler: (index: number, accountVote: AccountVote) => void;
-}): JSX.Element {
-  const { state } = context;
-  switch (state) {
-    case State.LOADING:
-      return <LoadingScreen />;
-    case State.STARTED: {
-      const { referenda, tracks, accountVotes } = context;
-      const referendumKeys = new Set(referenda.keys());
-      const voteKeys = new Set(accountVotes.keys());
-      if (referendumKeys.size > 0 && areEquals(referendumKeys, voteKeys)) {
-        // There are some referenda to vote on left
-        return (
-          <VotesTable
-            api={api}
-            connectedAccount={connectedAccount}
-            accountVotes={accountVotes}
-          />
-        );
+      );
+    case 'SyncedState': {
+      const { api, chain, votes } = state;
+      const { referenda, tracks } = chain;
+      const ongoingReferenda = filteringOngoingReferenda(referenda);
+      if (isVotingComplete(ongoingReferenda, votes)) {
+        // User went through all referenda
+        return <VotesSummaryTable api={api} accountVotes={votes} />;
```

Comment:
I believe this can be provided as a context that wraps the whole App component which then can replace the contexts/Api context.